webrtc-stats.html

```diff
@@ -2218,6 +2220,40 @@ <h2>
                   formats have ways to vary the quantizer value within the frame.
                 </p>
               </dd>
+              <dt>
+                <dfn>psnrSum</dfn> of type <span class="idlMemberType">
+                record&lt;DOMString, double&gt;</span>
+              </dt>
+              <dd>
+                <p>
+                  MUST NOT [= map/exist =] for audio.
+                  The cumulative sum of the PSNR values of frames encoded by this sender.
+                  The record includes values for the "y", "u" and "v" components.
+                  The count of measurements is in {{psnrMeasurements}}.
+                </p>
+                <p>
+                  PSNR is defined in [[ISO-29170-1:2017]].
+                </p>
+              </dd>
+              <dt>
+                <dfn>psnrMeasurements</dfn> of type <span class="idlMemberType">unsigned long long
+                </span>
+              </dt>
+              <dd>
+                <p>
+                  MUST NOT [= map/exist =] for audio.
+                  The number of times PSNR was measured. The components of {{psnrSum}} are
+                  aggregated with this measurement.
+                </p>
+                <p>
+                  The PSNR is defined in [[ISO-29170-1:2017]].
+                </p>
+                <p class="note">
+		  PSNR metrics should primarily be used as a basis for statistical analysis rather
+		  than be used as an absolute truth on a per-frame basis.
+                  The frequency of PSNR measurements is [=implementation-defined=].
+                </p>
```

Comment:
We want to avoid "should" and normative statements in non-normative notes. If we want to be normative we should lift it out.

Based on https://github.com/w3c/webrtc-stats/pull/794#issuecomment-2693542664 are we comfortable picking a min frequency of every 5 seconds?

```suggestion
                <p>
                  If the current encoder supports taking PSNR measurements, their
                  frequency SHOULD be no less than every 5 seconds.
                </p>
                <p class="note">
                  This allows for testing. PSNR measurements are intended for
                  statistical analysis, and aren't expected to be accurate down
                  to a frame.
                </p>
```

Reply:
Can you explain why 5 seconds? `keyFramesEncoded` and `keyFramesDecoded` are examples very-low-frequency events happening already where you can not make an assumption about the minimum interval between two getStats calls that will give you an increase. Same for packetsLost.

Reply:
What value would you like? The frequency of `keyFramesEncoded` and `keyFramesDecoded` are determined by external factors unlikely to vary by user agent.

Reply:
5 is arbitrary. In this thread I heard 15 seconds was too high, and that 1 second was not too expensive, but also that we'd rather not overconstrain implementations too much. 10 seconds?

This value, whatever we pick would go into WPT and also give web developers that can't wait for some reason a minimum time to wait to be interoperable.

Reply:
Hmm, maybe we need to also qualify that the encoder is actually encoding something? E.g. if the track is muted or a canvas track, then the frame rate may be less than what value we pick. Maybe we add _"... or the encoding frame rate whichever is lower"_?

Reply:
tl;dr: this is a parameter of the video encoder configuration. It becomes *observable* in stats just like others.
What the spec should say is "implementation defined" - the implied warning is "do not compare apples to oranges"

The frequency of keyFramesEncoded is a good example, it is determined by the encoder setting such as gop size. One wants this as large as possible for good performance but this is something where the encoder can be tuned without "interoperability" constraints since the decoder behavior is required to be flexible.

If one wants to get fancy about PSNR one may need to take into account things like "is this a screen sharing track" (higher resolution, lower frame rate and wholly different content) or frame rate (which may depend on BWE). Worth the effort for statistical analysis? Unlikely.

15 seconds was arbitrary too I think with the purpose of "i dont think this makes sense anymore". One second picked in the code is 3x the value from the paper (which has a parameter study) and hence 1/3rd as expensive in terms of power impact.

Reply:
A SHOULD implies implementation-defined. Is there a value that would be satisfactory?

Reply:
Moved the normative statement out of the note (and moved the note to the actual values)